fiftyone/utils/utils3d.py

```diff
@@ -774,16 +775,38 @@ def _clamp_to_discrete(arr, discrete):
     return clamp_list[np.clip(idx, 0, len(clamp_list) - 1)]
 
 
-# Reference: https://math.stackexchange.com/q/180418
+def _skew_symmetric_matrix(vec):
+    """Returns the skew-symmetric matrix of a 3D vector."""
+    return np.array(
+        [[0, -vec[2], vec[1]], [vec[2], 0, -vec[0]], [-vec[1], vec[0], 0]]
+    )
+
+
+# References:
+# 1. https://math.stackexchange.com/q/180418
+# 2. https://en.wikipedia.org/wiki/Rodrigues%27_rotation_formula#Matrix_notation
 def _rotation_matrix_from_vectors(vec1, vec2):
     """Returns the rotation matrix that aligns vec1 to vec2."""
     a = (np.asarray(vec1) / np.linalg.norm(vec1)).reshape(3)
     b = (np.asarray(vec2) / np.linalg.norm(vec2)).reshape(3)
     v = np.cross(a, b)
     c = np.dot(a, b)
-    s = np.linalg.norm(v)
-    K = np.array([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])
-    return np.eye(3) + K + K.dot(K) * ((1 - c) / (s**2))
+
+    if abs(c + 1) > np.finfo(c.dtype).eps:
+        K = _skew_symmetric_matrix(v)
+        return np.eye(3) + K + K.dot(K) / (1 + c)
```

Comment:
Nice, I had seen this simplification also

Reply:
Plug for https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.transform.Rotation.align_vectors.html

I use this `Rotation` module a lot myself in #3985

Reply:
If `scipy` is a valid dependency for this project (didn't even consider) than there's no point in having `_rotation_matrix_from_vectors` at all. We should just use `Rotation.align_vectors`. Less code to maintain and this PR can be reduced to a 1-2 LoC diff.

Reply:
`scipy` was added as a [dependency](https://github.com/voxel51/fiftyone/blob/develop/setup.py#L69) very very recently, and is not in any release yet. But it's already been implicitly required for a while, as we require `scikit-learn` and likely others that require `scipy`.

I would definitely be a big fan of using a module of its stature rather than rolling our own.
Please make this change, and if there are any other functions/operations that would more appropriately be replaced by `scipy`, let us know or add to this PR!